src/Couchbase.Lite.Shared/API/Sync/Replicator.cs

```diff
@@ -265,10 +266,97 @@ public void Stop()
             });
         }
 
+        /// <summary>
+        /// Gets a list of document IDs that are going to be pushed, but have not been pushed yet
+        /// <item type="bullet">
+        /// <description>API is a snapshot and results may change between the time the call was made and the time</description>
+        /// </item>
+        /// </summary>
+        /// <returns>An immutable set of strings, each of which is a document ID</returns>
+        /// <exception cref="CouchbaseLiteException">Thrown if no push replication</exception>
+        /// <exception cref="CouchbaseException">Thrown if an error condition is returned from LiteCore</exception>
+        [NotNull]
+        public IImmutableSet<string> GetPendingDocumentIDs()
+        {
+            if (!IsPushing()) {
+                WriteLog.To.Database.E(Tag, CouchbaseLiteErrorMessage.PullOnlyPendingDocIDs);
+                throw new CouchbaseLiteException(C4ErrorCode.Unsupported, CouchbaseLiteErrorMessage.PullOnlyPendingDocIDs);
+            }
+
+            if (_repl == null) {
+                StartInternal();
```

Comment:
This is not part of the technical spec, why is this starting the replicator?

Reply:
Actually mainly creating a replicator.... because test will fail when replicator is null..

Reply:
What test?  If the replicator is not started then this should simply return no documents.